File: library/src/main/java/me/yamlee/jsbridge/utils/SnackBarUtils.java
package me.yamlee.jsbridge.utils;

import android.support.design.widget.Snackbar;
import android.view.View;
import android.widget.TextView;

import me.yamlee.jsbridge.R;


/**
 * 底部提示
 * <p>
 *
 * @author zcZhang
 * @author yamlee
 */
public class SnackBarUtils {
    public static void showLongSnackBar(View view, String msg, String actionTitle) {

        Snackbar.make(view, msg, Snackbar.LENGTH_LONG)
                .setAction(actionTitle, new View.OnClickListener() {
                    @Override
                    public void onClick(View v) {

                    }
                }).show();
    }

    public static void showShortSnackBar(View view, String msg, String actionTitle) {
        Snackbar snackbar = Snackbar.make(view, msg, Snackbar.LENGTH_SHORT);
        View snackbarView = snackbar.getView();
        TextView textView = (TextView) snackbarView.findViewById(android.support.design.R.id.snackbar_text);
        textView.setMaxLines(5);  // show multiple line
        snackbar.setAction(actionTitle, new View.OnClickListener() {
            @Override
            public void onClick(View v) {

            }
        }).show();
    }

    public static void showClickSnackBar(View view, String msg, String actionTitle,
                                         View.OnClickListener clickListener) {
        Snackbar.make(view, msg, Snackbar.LENGTH_LONG)
                .setAction(actionTitle, clickListener).show();
    }

    public static void showClickSnackBarAlways(View view, String msg, String actionTitle,
                                               View.OnClickListener clickListener) {
        Snackbar.make(view, msg, Snackbar.LENGTH_LONG)
                .setAction(actionTitle, clickListener).show();
    }

    public static void showShortSnackBar(View view, String msg) {
        showShortSnackBar(view, msg, view.getContext().getString(R.string.i_know_it));
    }
}

File: android-jsBridgeAll/src/main/java/me/yamlee/bridge/ui/util/SnackBarUtils.kt
package me.yamlee.bridge.ui.util

import android.support.design.widget.Snackbar
import android.view.View
import android.widget.TextView

import me.yamlee.bridge.ui.R


/**
 * 底部提示
 *
 *
 *
 * @author zcZhang
 * @author yamlee
 */
object SnackBarUtils {
    fun showLongSnackBar(view: View, msg: String, actionTitle: String) {

        Snackbar.make(view, msg, Snackbar.LENGTH_LONG)
                .setAction(actionTitle) { }.show()
    }

    @JvmOverloads
    fun showShortSnackBar(view: View, msg: String, actionTitle: String = view.context.getString(R.string.i_know_it)) {
        val snackbar = Snackbar.make(view, msg, Snackbar.LENGTH_SHORT)
        val snackbarView = snackbar.view
        val textView = snackbarView.findViewById<View>(android.support.design.R.id.snackbar_text) as TextView
        textView.maxLines = 5  // show multiple line
        snackbar.setAction(actionTitle) { }.show()
    }

    fun showClickSnackBar(view: View, msg: String, actionTitle: String,
                          clickListener: View.OnClickListener) {
        Snackbar.make(view, msg, Snackbar.LENGTH_LONG)
                .setAction(actionTitle, clickListener).show()
    }

    fun showClickSnackBarAlways(view: View, msg: String, actionTitle: String,
                                clickListener: View.OnClickListener) {
        Snackbar.make(view, msg, Snackbar.LENGTH_LONG)
                .setAction(actionTitle, clickListener).show()
    }
}
